### PR TITLE
Cleanup on ticker allocations.

### LINKED ISSUE #10596: auto-pause: make it memory-leak proof
each call of `time.After` creates a new timer instance which is not released until the timer fired. If we have high traffic in `incomeCh` we'll end up with a lot of pending timers

https://groups.google.com/g/golang-nuts/c/cCdm0Ixwi9A

_Originally posted by @ilya-zuyev in https://github.com/kubernetes/minikube/pull/10427#discussion_r580669849_

https://www.fatalerrors.org/a/use-with-caution-time.after-can-cause-memory-leak-golang.html

### PR DESCRIPTION
**Pull request by vishjain**
fixes #10596

### CODE REVIEW

**Comment by vishjain** [comment]
Here is the outout: 
minikube vishal$ minikube addons enable auto-pause

❗  Executing "docker container inspect minikube --format={{.State.Status}}" took an unusually long time: 2.611768229s
💡  Restarting the docker service may improve performance.
    ▪ Using image gcr.io/k8s-minikube/auto-pause-hook:v0.0.2
    ▪ auto-pause addon is an alpha feature and still in early development. Please file issues to help us make it better.
    ▪ https://github.com/kubernetes/minikube/labels/co/auto-pause

File: cmd/auto-pause/auto-pause.go [hunk 1]
```diff
@@ -48,21 +48,38 @@ func main() {
 	// TODO: #10595 make this configurable
 	const interval = time.Minute * 1
 
+	// Check if interval is greater than 0 so NewTicker does not panic.
+	if interval <= 0 {
+		exit.Message(reason.Usage, "Auto-pause interval must be greater than 0,"+
+			" not current value of {{.interval}}", out.V{"interval": interval.String()})
+	}
+	tickerChannel := time.NewTicker(interval)
```
Comment (andriyDev) [inline cmd/auto-pause/auto-pause.go:56]: defer stop the ticker.

Comment (vishjain) [reply cmd/auto-pause/auto-pause.go:56]: Putting a defer stop the ticker give a linter error: cmd/auto-pause/auto-pause.go:93:2: exitAfterDefer: log.Fatal will exit, and `defer tickerChannel.Stop()` will not run (gocritic)
        log.Fatal(http.ListenAndServe("0.0.0.0:8080", nil))
        ^
make: *** [lint] Error 1

Comment (vishjain) [reply cmd/auto-pause/auto-pause.go:56]: Instead, I'll put it at the very end.

**Comment by vishjain** [comment]
@andriyDev can you please review this? Thanks!

**Review by medyagh — APPROVED** [review]
thanks @vishjain

**Pull request merged**